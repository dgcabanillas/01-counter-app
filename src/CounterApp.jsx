import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState( value );

    const handleAdd = () => setCounter( counter + 1 );
    const handleReset = () => setCounter( value );
    const handleSubtract = () => setCounter( counter - 1 );

    return (  
        <Fragment>
            <h1> CounterApp </h1>
            <h2>{ counter }</h2>
            <button onClick={ handleSubtract }> - </button>
            <button onClick={ handleReset }> Reset </button>
            <button onClick={ handleAdd }> + </button>
        </Fragment>
    );
}
 
CounterApp.propTypes = {
    value: PropTypes.number
}
CounterApp.defaultProps = {
    value: 0
}
export default CounterApp;