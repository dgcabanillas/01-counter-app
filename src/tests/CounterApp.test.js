import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../../src/CounterApp';

describe('Pruebas en <CounterApp />', () => {

    let wrapper = shallow( <CounterApp /> );

    beforeEach( () => {
        wrapper = shallow( <CounterApp /> );
    });

    test('Debe mostrar el componente <CounterApp /> de manera correcta', () => {
        const value = wrapper.find('h2').text();
        expect( wrapper ).toMatchSnapshot();
        expect( value ).toBe( "0" );
    });

    test('Debe mostrar el componente <CounterApp /> de manera correcta con parametro', () => {
        wrapper = shallow( <CounterApp value = { 100 } /> );
        const counterText = wrapper.find('h2').text();
        expect( wrapper ).toMatchSnapshot();
        expect( counterText ).toBe( "100" );
    });

    test('Debe decrementar con el boton -1 btn[0]', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect( counterText ).toBe("-1"); 
    });

    test('Debe de incrementar con el boton +1 btn[2]', () => {
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect( counterText ).toBe("1"); 
    });

    test('Debe de resetear el valor a 0 con el boton reset btn[1]', () => {
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect( counterText ).toBe("0"); 
    });

    test('Debe de resetear el valor al valor por defecto con el boton reset btn[1]', () => {
        wrapper = shallow( <CounterApp value={200} /> );
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect( counterText ).toBe("200"); 
    });

});