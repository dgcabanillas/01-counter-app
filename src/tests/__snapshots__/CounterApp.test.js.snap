// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe mostrar el componente <CounterApp /> de manera correcta 1`] = `
<Fragment>
  <h1>
     CounterApp 
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
     - 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    onClick={[Function]}
  >
     + 
  </button>
</Fragment>
`;

exports[`Pruebas en <CounterApp /> Debe mostrar el componente <CounterApp /> de manera correcta con parametro 1`] = `
<Fragment>
  <h1>
     CounterApp 
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
     - 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    onClick={[Function]}
  >
     + 
  </button>
</Fragment>
`;
